HelpSource/Classes/Process.schelp

```diff
@@ -56,3 +58,235 @@ list::
 This means that link::Classes/Thread#.thisThread#thisThread:: will always initially point
 to the main Thread. However, when some code starts a link::Classes/Routine::, the Routine
 becomes the current Thread, with the main Thread as its parent.
+
+examples::
+
+strong::Example 1.:: Comparison of the path of the evaluated code block in a saved SCD, the loaded SCD and the function in the loaded SCD:
+
+table::
+##
+
+If a code block ("fileMain.scd" in the example code below) executes another file on disk ("fileForLoad.scd" in the example code below), using link::Classes/String#-load:: or link::Classes/String#-loadPaths::, teletype::thisProcess.nowExecutingPath:: will be the location of the executed file ("fileForLoad.scd" in the example code below). However, if the function is called in the executed file ("fileForLoad.scd" in the example code below), teletype::thisProcess.nowExecutingPath:: will be the document of the code block ("fileMain.scd" in the example code below).
```

Comment:
I'm confused - if a function containing `thisProcess.nowExecutingPath` is called, `thisProcess.nowExecutingPath` will return the path of the document containing the function call, right? You are defining the function in fileForLoad but calling it in fileMain...

Reply:
Ok, I got confused now too.

Reply:
Maybe deserves a less confusing rewrite? Writing like your comment "path of the document containing the function call" is already more clear.

Reply:
Oh, sorry! You both are correct. My writing is wrong.

What I wanted to write was:

>However, if the function stored in the executed file ("fileForLoad.scd" in the example code below) is called  in another file ("fileMain.scd" in the example code below), `thisProcess.nowExecutingPath` will be the
document of the code block ("fileMain.scd" in the example code below).

How about the following?

>If a code ("fileMain.scd" in the example code below) executes another file on disk ("fileForLoad.scd" in the example code below) using `String: -load` or `String: -loadPaths`, `thisProcess.nowExecutingPath` will be the location of the executed file ("fileForLoad.scd" in the example code below). If a function containing `thisProcess.nowExecutingPath` is called (the function defined in "fileForLoad.scd" in the example code below), `thisProcess.nowExecutingPath` will return the path of the document containing the function call ("fileMain.scd" in the example code below).

Reply:
@cdbzb @smoge 
Thanks and sorry! You both are correct. My writing is wrong.

What I wanted to write was:

>However, if the function stored in the executed file ("fileForLoad.scd" in the example code below) is called in another file ("fileMain.scd" in the example code below), `thisProcess.nowExecutingPath` will be the document of the code block ("fileMain.scd" in the example code below).

(I seem to have left some words out while writing)


@smoge. Yes, @cdbzb's sentence is very clear! Thanks!
How about the following?

>If a code ("fileMain.scd" in the example code below) executes another file on disk ("fileForLoad.scd" in the example code below) using `String: -load` or `String: -loadPaths`, `thisProcess.nowExecutingPath` will be the location of the executed file ("fileForLoad.scd" in the example code below). If a function containing `thisProcess.nowExecutingPath` is called (the function defined in "fileForLoad.scd" in the example code below), `thisProcess.nowExecutingPath` will return the path of the document containing the function call ("fileMain.scd" in the example code below).

Reply:
It works different with this example:

```
// Main.scd
thisProcess.nowExecutingPath.postln ;
~func = {thisProcess.nowExecutingPath.postln};
"~/test/secondary.scd".standardizePath.load;

// secondary.scd
thisProcess.nowExecutingPath.postln;
~func.value.postln;

// OUTPUT
/Users/smoge/test/main.scd
/Users/smoge/test/secondary.scd
/Users/smoge/test/secondary.scd
/Users/smoge/test/secondary.scd
```

Reply:
It's always the path containing the function call, right?

Reply:
Is it necessary to create other files in the help file to demonstrate this? I'm not sure it will just take the focus away from the main point. What do you think?

Reply:
@smoge 

>// Main.scd
>
>thisProcess.nowExecutingPath.postln ;
>~ func = {thisProcess.nowExecutingPath.postln};
"~/test/secondary.scd".standardizePath.load;
~func2.value;
>
>// secondary.scd
thisProcess.nowExecutingPath.postln;
{~func}.value.value.postln;
~func2 = {thisProcess.nowExecutingPath.postln};
>
>// OUTPUT
/Users/smoge/test/main.scd
/Users/smoge/test/secondary.scd
/Users/smoge/test/secondary.scd
/Users/smoge/test/secondary.scd
/Users/smoge/test/main.scd


Yes!
By evaluating the example in this PR:
- fileMain.scd:
   ```
   (
   ("thisProcess.nowExecutingPath:").postln;
   ("	in fileMain.scd:                       " + thisProcess.nowExecutingPath).postln;
   ~test = (thisProcess.nowExecutingPath.dirname +/+ "fileForLoad.scd");
   ~test.load.testFunction;
   ~test.openOS; // n.b.: .openDocument only works in SC-IDE
   'fileMain.scd tasks finished'.postln;
   )
   ```
- fileForLoad.scd:
   ```
   ("	in fileForLoad.scd:                    " + thisProcess.nowExecutingPath).postln;
   (
   testFunction: {
   	("	in the testFunction in fileForLoad.scd:" + thisProcess.nowExecutingPath).postln
   }
   )
   ```
- Post window:
   ```
   -> a File
   -> /Users/prko/fileMain.scd
   thisProcess.nowExecutingPath:
   	in fileMain.scd:                        /Users/prko/fileMain.scd
   	in fileForLoad.scd:                     /Users/prko/fileForLoad.scd
   	in the testFunction in fileForLoad.scd: /Users/prko/fileMain.scd
   fileMain.scd tasks finished
   -> fileMain.scd tasks finished
   ```

>Is it necessary to create other files in the help file to demonstrate this? I'm not sure it will just take the focus away from the main point. What do you think?

Evaluating a path in HelpBrowser returns a wrong path. To make it easy for readers to understand how it works, users should learn it by doing or by watching a video. I think including this example helps. For me, http://doc.sccode.org/Classes/File.html is a well-written example in this respect.

Reply:
I'm not sure what this line demonstrates:

`~test = (thisProcess.nowExecutingPath.dirname +/+ "fileForLoad.scd");`